### DISCUSSION TITLE
How do modify joint limits temporarily for one planning query

### DISCUSSION THREAD

**Discussion by peter-mitrano-bg** [NVlabs/curobo#261] [Software Q&A]
I'd like to modify either the input yml, dict, RobotConfig, or MotionGen config some way to update the joint limits without permenantly modifying the URDF on disc. I see `_update_joint_limits` in `cuda_robot_generator` but that seems to be private, and probably can't be called after the `_kinematics_config` has been constructed.

So is there a way I can do this?

**Comment by xiaomengfang** [comment]
Hi, there, have you solved this problem?

**Comment by KySmith1** [comment]
@balakumar-s - bumping this for awareness as it's a feature we're looking for too. [Similar question](https://github.com/NVlabs/curobo/discussions/376) was asked back in August.

Is there any issue / alternative path to us updating joint limits as seen above? Making multiple configs of URDFs to communicate temporary joint constraints is a fine pattern for us.
